web-ng/src/app/services/auth/unauth.guard.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { AuthService } from './auth.service';
+import { Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { environment } from '../../../environments/environment';
+
+const DEFAULT_ROUTE = ['p', ':new'];
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UnauthGuard {
```

Comment:
I did some searching and found that `canActivate()` can take an  [`ActivatedRouteSnapshot`](https://angular.io/api/router/ActivatedRouteSnapshot). In that case, instead of creating a separate guard class as per my original brainstorm (sorry!), can we just instead just update AuthGuard to react differently when trying to activate `/signin`? 

In the future if we have more routes we can make these behaviors configurable by passing `data` as part of the Routers in `RoutingModule`; for now hard-coding `/signin` in the AuthGuard should be ok.